homework/src/index.js

```diff
@@ -1,47 +1,126 @@
+        const li = createAndAppend('li', ul);
+
+        const contributorsList = fetchJSON(contributors_url)
+
+        // console.log(contributorsList);  
+        contributorsList.then(data => {
+            data.forEach((item) => {
+
+                const link_to_repo = createAndAppend('a', li);
+                link_to_repo.setAttribute("href", item.html_url);
+                link_to_repo.setAttribute("target", "_blank");
+
+                const avatar = createAndAppend('img', link_to_repo, { class: 'avatar' });
+                avatar.setAttribute("src", item.avatar_url);
+                avatar.setAttribute("width", "60");
+                createAndAppend('p', li, { text: item.login });
+                createAndAppend('p', li, { text: item.contributions, class: 'contributors_number' });
+                createAndAppend('li', li, { class: 'line', });
+            });
+
+        });
+
     };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'html') {
-        elem.innerHTML = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
-
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-
-  window.onload = () => main(HYF_REPOS_URL);
-}
+
+    function main(url) {
+        const root = document.getElementById('root');
+
+        const div = createAndAppend('div', root, { class: '' });
+        const header = createAndAppend('h3', div, { text: "HYF Repositories  ", class: 'header' });
+        const select = createAndAppend('select', header, { class: "select" });
+        const container = createAndAppend('div', root, { class: "container" });
+
+        fetchJSON(url)
+            .then(data => {
+                data.sort(function(a, b) {
+                    return a.name.localeCompare(b.name)
+                })
+                data.forEach((item, index) => {
+                    createAndAppend('option', select, { text: item.name, value: index });
+                });
+
+                select.addEventListener('change', (event) => {
+                    const index = event.target.value;
+
+
+                    renderRepo(container, data[index].html_url, data[index].description, data[index].forks, data[index].updated_at, select[index].textContent);
```

Comment:
As @JawharBairakdar indicated, with respect to readability, there are too many argument passed here. You can just pass `data[index]` (or preferabl, rename `data` to `repositories` so that you can pass `repositories[index]`). So the `renderRepo` function itself could be defined as:

```js
function renderRepo(container, repository)
```